crates/cxx-qt-gen/src/generator/rust/constructor.rs

```diff
@@ -0,0 +1,310 @@
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #initialize_arguments_cxx]
+                #[doc(hidden)]
+                struct #initialize_arguments_rust {
+                    #(#initialize_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
+                                 // doesn't support empty shared structs.
+                }
+            },
+            parse_quote! {
+                extern "Rust" {
+                    #[namespace = #namespace_internals]
+                    #[cxx_name = #route_arguemnts_cxx]
+                    // This function needs to marked unsafe, as some arguments may be pointers.
+                    unsafe fn #route_arguments_rust(#(#route_arguments_parameters),*) -> #arguments_rust;
+
+                    #[namespace = #namespace_internals]
+                    #[cxx_name = #new_cxx]
+                    fn #new_rust(args: #new_arguments_rust) -> Box<#rust_struct_name_rust>;
+
+                    #[namespace = #namespace_internals]
+                    #[cxx_name = #initialize_cxx]
+                    fn #initialize_rust(qobject: Pin<&mut #qobject_name_rust>, args: #initialize_arguments_rust);
+                }
+            },
+        ]);
+        result.cxx_qt_mod_contents.append(&mut vec![parse_quote! {
+            #[doc(hidden)]
+            pub fn #route_arguments_rust(#(#route_arguments_parameters),*) -> #arguments_rust {
+                // these won't be used if any of them don't have any members
+                #[allow(unused_variable)]
+                let (
+                    new_arguments,
+                    base_arguments,
+                    initialize_arguments
+                    ) = <#qobject_name_rust as cxx_qt::Constructor<(#(#argument_types,)*)>>
+                                    ::route_arguments((#(#assign_arguments,)*));
+                #arguments_rust {
+                    baseArguments: #base_arguments_rust {
+                        #(#init_base_arguments,)*
+                        notEmpty: 0
+                    },
+                    initializeArguments: #initialize_arguments_rust {
+                        #(#init_initialize_arguments,)*
+                        notEmpty: 0
+                    },
+                    newArguments: #new_arguments_rust {
+                        #(#init_new_arguments,)*
+                        notEmpty: 0
+                    },
+                }
+            }
+        },
+        parse_quote! {
+            #[doc(hidden)]
+            pub fn #new_rust(new_arguments: #new_arguments_rust) -> Box<#rust_struct_name_rust> {
+                let new_arguments = (#(#extract_new_arguments,)*);
+                Box::new(<#qobject_name_rust as cxx_qt::Constructor<(#(#argument_types,)*)>>::new(new_arguments))
```

Comment:
do you need `alloc::boxed::Box` here for both the Boxes to fully qualify them like elsewhere?